### PR TITLE
Get last events

### PR DESCRIPTION
**Pull request by sakisk**
Get last 3 events from meetup api to have available at feedback page.

Talks and speakers are extracted from first line of event's description

### CODE REVIEW

**Comment by sakisk** [comment]
:shipit:

**Comment by kevbite** [comment]
Seem to have some failing tests, mind taking a look - https://ci.appveyor.com/project/kevbite/dotnetsheff-api/builds/22538319/tests

Thanks

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/GetAvailableFeedbackEvents.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            var talkParsers = Container.Instance.Resolve<IEnumerable<ITalkParser>>(log).ToArray();
+
+            var eventTalks = new List<EventTalks>();
+
+            foreach (var lastEvent in lastEvents)
+            {
+                foreach (var parser in talkParsers)
+                {
+                    var talks = parser.Parse(lastEvent).ToArray();
+                    if (talks.Any())
```
Comment (kevbite) [inline src/dotnetsheff.Api/GetAvailableFeedbackEvents/GetAvailableFeedbackEvents.cs:36]: Remove this branching with
```csharp
eventTalks.AddRange(talks);
````

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/ITalkParser.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+using System.Collections.Generic;
+
+namespace dotnetsheff.Api.GetAvailableFeedbackEvents
+{
+    public interface ITalkParser
+    {
+        IEnumerable<Talk> Parse(PastEvent pastEvent);
+    }
+
+    public class Talk
```
Comment (kevbite) [inline src/dotnetsheff.Api/GetAvailableFeedbackEvents/ITalkParser.cs]: A separate file, please.

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace dotnetsheff.Api.GetAvailableFeedbackEvents
+{
+    public class OneSpeakerTwoTalksParser : ITalkParser
+    {
+        public IEnumerable<Talk> Parse(PastEvent pastEvent)
+        {
+            var oneSpeakerRegex = new Regex("This event will be (split|broken down) into (2|two) (parts|talks), (?<talk1>.+) and (?<talk2>.+) both being presented by (?<speaker>.+).");
```
Comment (kevbite) [inline src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs]: Why does `(?<speaker>.+).` have a `.` (any char) at the end of that group?

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+            var match = oneSpeakerRegex.Match(pastEvent.Description);
+
+            if (!match.Success) yield break;
```
Comment (kevbite) [inline src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs:14]: This all flows nicely, I like it 👍

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/TwoSpeakersTalkParser.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace dotnetsheff.Api.GetAvailableFeedbackEvents
+{
+    public class TwoSpeakersTalkParser : ITalkParser
+    {
+        public IEnumerable<Talk> Parse(PastEvent @event)
+        {
+            var twoSpeakersRegex = new Regex("This event will be split into two parts, (?<speaker1>.+) presenting (?<talk1>.+) and the second half will be (?<speaker2>.+) presenting (?<talk2>.+).");
```
Comment (kevbite) [inline src/dotnetsheff.Api/GetAvailableFeedbackEvents/TwoSpeakersTalkParser.cs]: Same again with the "any" on the grouping `(?<talk2>.+).`?

File: src/dotnetsheff.Api/Meetup/IMeetupApi.cs [hunk 6]
```diff
@@ -13,5 +14,8 @@ public interface IMeetupApi
 
         [Get("/{group}/events")]
         Task<List<AlexaSkillEvent>> GetAlexaSkillEventsAsync(string group, string apiKey, string status, string page, string omit);
+
+        [Get("/{group}/events")]
```
Comment (kevbite) [inline src/dotnetsheff.Api/Meetup/IMeetupApi.cs:18]: Magic of `Refit` 🤮

File: src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs (hunk 3 above)
Comment (sakisk) [reply src/dotnetsheff.Api/GetAvailableFeedbackEvents/OneSpeakerTwoTalksParser.cs]: regex matches the `.` when speaker name at end of sentence

**Comment by kevbite** [comment]
Looks like you've missed out Lightning talks?

File: src/dotnetsheff.Api/Meetup/IMeetupApi.cs (hunk 6 above)
Comment (sakisk) [reply src/dotnetsheff.Api/Meetup/IMeetupApi.cs:18]: team america vomit

**Comment by sakisk** [comment]
> Looks like you've missed out Lightning talks?

we need a parser for those?

**Pull request merged**